contracts/IdleCDOCardManager.sol

```diff
@@ -37,56 +40,78 @@ contract IdleCDOCardManager is ERC721Enumerable {
     return idleCDOs;
   }
 
-  function mint(address _idleCDOAddress, uint256 _risk, uint256 _amount) public returns (uint256) {
-    // check if _idleCDOAddress exists in idleCDOAddress array
-    require(isIdleCDOListed(_idleCDOAddress), "IdleCDO address is not listed in the contract");
+  function mint(
+    address _idleCDOPos1Address,
+    uint256 _riskPos1,
+    uint256 _amountPos1,
+    address _idleCDOPos2Address,
+    uint256 _riskPos2,
+    uint256 _amountPos2
+  ) external returns (uint256) {
+    require(_amountPos1 > 0 || _amountPos2 > 0, "cannot mint with no amount");
 
-    IdleCDOCard _card = new IdleCDOCard(_idleCDOAddress);
-    IERC20Detailed underlying  = IERC20Detailed(IdleCDO(_idleCDOAddress).token());
+    // mint the Idle CDO card
+    _tokenIds.increment();
+    uint256 tokenId = _tokenIds.current();
+    _mint(msg.sender, tokenId);
 
-    // transfer amount to cards protocol
-    underlying.safeTransferFrom(msg.sender, address(this), _amount);
+    IdleCDOCard _card = new IdleCDOCard();
 
-    // approve the amount to be spend on cdos tranches
-    underlying.approve(address(_card), _amount);
+    if (_amountPos1 > 0) {
+      // deposit position 1
+      _depositToCard(_card, _idleCDOPos1Address, _riskPos1, _amountPos1);
 
-    // calculate the amount to deposit in BB
-    // proportional to risk
-    uint256 depositBB = percentage(_risk, _amount);
+      Card memory newCard;
```

Comment:
an improved version of this could be
```
      uint256 _currId = _cardIds.current();
      _cardMap[_currId] = Card(_riskPos1, _amountPos1, address(_card), _idleCDOPos1Address);
      _cards[tokenId].push(_currId);
      _cardIds.increment();
```
so you save 1 variable for `newCard` and 1 SLOAD for `_cardIds.current()`

Reply:
```_cardIds.current(); ``` was put into a variable to avoid 2 SLOADS